test/tests/StringTest.cpp

```diff
@@ -64,3 +64,25 @@ TEST_F(StringTest, Split_ByEmpty)
 {
     EXPECT_THROW(String::Split("string", ""), std::invalid_argument);
 }
+
+TEST_F(StringTest, ToUpper_Basic)
+{
+    utf8 *actual = (utf8*)alloca(20 * sizeof(utf8));
```

Comment:
Any reason for using `alloca` over `utf8[]`? The compiler knows the size already.

Reply:
Nope, good catch.